pkg/cluster/cluster_info_scraper.go

```diff
@@ -326,11 +326,18 @@ func (s *ClusterScraper) GetResourcesPaginated(
 }
 
 func (s *ClusterScraper) GetKubernetesServiceID() (svcID string, err error) {
+	if cachedSvcID, exists := s.cache.Get(kubernetesServiceName); exists {
```

Comment:
Just a nit, may be use `fmt.Sprintf("%v-%v", k8sDefaultNamespace, kubernetesServiceName)` as the key? Maybe consider adding a function in `key_func.go`.

Reply:
I agree, I wasn't sure that just service name would be a good value for a key but thank you for bringing this